github_slack_bot/utils/slack.py

```diff
@@ -0,0 +1,145 @@
+import logging
+import re
+from slackclient import SlackClient
+from slackclient.client import SlackNotConnected
+
+# TODO: Allow log level to be passed in as an application parameter
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d:%(funcName)s] %(message)s",
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+RTM_READ_INTERVAL = 1  # Number of seconds to wait between each rtm_read() call
+EXCEPTION_DEFAULT_MSG = "Something unexpected occurred, see ERROR log and Traceback above"
+
+
+class SlackBot:
+    """The SlackBot listens to events from the Slack RTM API and creates GitHub issues
+    from messages that mention the bot's username.
+    """
+
+    # TODO: extend is_bot_message() to allow bot mentions anywhere in the message
+    @staticmethod
+    def is_bot_message(bot_user_id, message):
+        """Check if the message mentions the bot_user_id at
+        the start of the message and includes a body
+        :param bot_user_id: bot's user ID to search for
+        :param message: message sent by user
+        :return: True if bot is mentioned and message contains body, False otherwise
+        """
+        message_stripped = message.strip()
+        logger.debug("Checking if bot '{}' is mentioned in message: '{}'"
+                     .format(bot_user_id, message_stripped))
+        return message_stripped.startswith("<@{}> ".format(bot_user_id))
+
+    # TODO: give user ability to provide a GitHub repo in their message body
+    # TODO: add ability to add assignees,labels, projects, milestones to Issues
+    @staticmethod
+    def parse_bot_message(msg):
+        """Parse a message mentioning our bot
+        :param msg: message sent by user
+        :return: A dict with the issue title and body if they exist,
+        a dict with an error message otherwise
+                Example:
+                {"ok": True, "issue_title": "Issue title here", "issue_body": "Issue body here"}
+                {"ok": False, "error": "Error message goes here"}
+        """
+        msg_regex = "^<@(.+?)> (.*)"
+        matches = re.search(msg_regex, msg)
+        msg_body = matches.group(2).strip()
+        msg_body_split = msg_body.split("__")
+        logger.debug("Split message into {}".format(msg_body_split))
+
+        if len(msg_body_split) == 2:
+            issue_title = msg_body_split[0]
+            issue_body = msg_body_split[1]
+            res = {
+                "ok": True,
+                "issue_title": "{}".format(issue_title),
+                "issue_body": "{}".format(issue_body)
+            }
+        else:
+            res = {
+                "ok": False,
+                "error": "Failed to parse message. Issue title and issue body should be "
+                         "separated by a double underscore (__)"
+            }
+
+        logger.info(res)
+        return res
+
+    def __init__(self, slack_api_token=None, git_client=None):
```

Comment:
Why are `slack_api_token` and `git_client` optional here? If they're not passed, I think the program will raise an exception later when slack bot attempts to use them but they'll be None.

Reply:
I think I left them that way from when I first started testing, will change them to required params.